src/irc/IrcEventBroker.ts

```diff
@@ -361,6 +361,12 @@ export class IrcEventBroker {
                 req, server, createUser(nick), chan, "part", reason
             ));
         });
+        this.hookIfClaimed(client, connInst, "kick", (chan: string, nick: string, by: string, reason: string) => {
+            const req = createRequest();
+            complete(req, ircHandler.onKick(
+                req, server, createUser(by), createUser(nick), chan, reason
+            ));
+        });
         this.hookIfClaimed(client, connInst, "quit", (nick: string, reason: string, chans: string[]) => {
```

Comment:
I moved the quit function to be near the join function so that it would appear

Reply:
clever